src/server/http/__tests__/short_url_error.js

```diff
@@ -2,24 +2,37 @@ import expect from 'expect.js';
 import _ from 'lodash';
 import { handleShortUrlError } from '../short_url_error';
 
+function createErrorWithStatus(status) {
+  const error = new Error();
+  error.status = status;
+  return error;
+}
+
+function createErrorWithStatusCode(statusCode) {
+  const error = new Error();
+  error.statusCode = statusCode;
+  return error;
+}
+
 describe('handleShortUrlError()', () => {
-  const caughtErrors = [{
-    status: 401
-  }, {
-    status: 403
-  }, {
-    status: 404
-  }];
+  const caughtErrors = [
+    createErrorWithStatus(401),
+    createErrorWithStatus(403),
+    createErrorWithStatus(404),
+    createErrorWithStatusCode(401),
+    createErrorWithStatusCode(403),
+    createErrorWithStatusCode(404),
+  ];
 
-  const uncaughtErrors = [{
-    status: null
-  }, {
-    status: 500
-  }];
+  const uncaughtErrors = [
+    new Error(),
+    createErrorWithStatus(500),
+    createErrorWithStatusCode(500)
+  ];
 
   caughtErrors.forEach((err) => {
-    it(`should handle ${err.status} errors`, function () {
-      expect(_.get(handleShortUrlError(err), 'output.statusCode')).to.be(err.status);
+    it(`should handle ${err.status || err.statusCode} errors`, function () {
+      expect(_.get(handleShortUrlError(err), 'output.statusCode')).to.be(err.status || err.statusCode);
```

Comment:
The first error in `uncaughtErrors` is going to have a status and statusCode of undefined, right?